protocol/server.go

```diff
@@ -179,8 +186,8 @@ func (s *Server) String() string {
 }
 
 func (s *Server) logf(format string, args ...interface{}) {
```

Comment:
why not just default s.trace to this:
func (s *Server) logf(strings, ...interface{}) {}

Then if you want a logger, you can s.trace to  a function.

That way, for logging, you're only ever calling one function, not potentially two.

Reply:
The old code had a lot of checks like:
```go
if s.trace != nil {
    s.trace("write back %v,", m.Type) 
}
```
Seemed cleaner to have wrap it in a function but it would add an extra call.